pkg/client/client.go

```diff
@@ -347,22 +365,22 @@ func checkAwaitingKeyGenerationForKeep(
 	// - public key submission transactions are still mining,
 	// - conflicting public key has been submitted.
 	// In both cases, the client should not attempt to generate the key again.
-	if keepsRegistry.HasSigner(keep) {
+	if keepsRegistry.HasSigner(keep.ID()) {
 		logger.Warningf(
 			"keep public key is not registered on-chain but key material "+
 				"is stored on disk; skipping key generation; PLEASE INSPECT "+
 				"PUBLIC KEY SUBMISSION TRANSACTION FOR KEEP [%v]",
```

Comment:
I think we need to change `%v` to `%s` here now so that `String()` is called under the hood.